src/js/load-package.ts

```diff
@@ -276,6 +276,33 @@ export class PackageManager {
     }
   }
 
+  public async loadBinaryPackage(
+    name: string,
+    buffer: ArrayBuffer | Uint8Array,
+    format?: string,
+  ) {
+    const metadata: PackageLoadMetadata = {
+      name,
+      normalizedName: '---',
+      channel: this.defaultChannel,
+      depends: [],
+      done: createResolvable(),
+      installPromise: undefined,
+      packageData: {
+        name,
+        version: '1.0.0',
+        file_name: name + '-1.0.0-py3-none-any.' + (format || 'whl'),
+        install_dir: 'site',
+        sha256: '',
+        package_type: 'package',
+        imports: [],
+        depends: []
+      }
+    };
+
+    await this.installPackage(metadata, new Uint8Array(buffer));
```

Comment:
If `buffer` is a `Uint8Array` this copies. You want `API.typedArrayAsUint8Array` https://github.com/pyodide/pyodide/blob/main/src/core/pre.js?plain=1#L86

Reply:
great suggestion! ..thanks

in the process of removing an earlier commit belonging to a different PR..
I (effectively) rebased to the current HEAD of main.
You guys move quickly :)